### ISSUE TITLE
Show existing POIs on long-press of location

### ISSUE THREAD

**Issue by IzzySoft** [streetcomplete/StreetComplete#4253]
**Use case**
I'm out-and-about solving my quests and wonder about why this X I see in front of my nose does not show up. Sure, I can simply add a hint – but how do I know that this does not just "create noise" and the X I want to add is already there?

**Proposed Solution**
When long-pressing on a location, I currently get 3 choices: Create a new hint, start a track recording, open location in other app. It would be nice to have an option like on the OSM website: "object query" or, in simple terms, "What is here?" – to simply show all existing "objects" at the selected position (and a sane radius around it). That way I can…

* see if X was already mapped
* check the properties of X if they are correct
* e.g. add opening times to a shop or restaurant if it does not yet have any, even if there was no quest for it

Before someone asks: no, I don't expect "all sophisticated details" of a full OSM editor. What we all love about StreetComplete is the simplicity, enabling even non-tech-savvy users to contribute – and this should remain. So actions should orient on this, allowing a similar way of processing as the quests do. My main concern is seeing whether a thing is already mapped, without having to jump between multiple apps I'd need to have installed for that – and to contribute to it if needed/possible in the same easy way SC provides with its quests.

And while I probably could use "open location in other app" to open it with e.g. OsmAnd, most users would feel that to complicated for *editing* that point then. I'd welcome if simple things like checking/fixing/adding opening times would be possible directly from within StreetComplete, even if there's no current quest.

**Comment by mnalis** [comment]
Somewhat related: https://github.com/streetcomplete/StreetComplete/issues/471 and discussion https://github.com/streetcomplete/StreetComplete/discussions/3003.

Also note that keeping the map clear as possible and not showing all POIs is on purpose (e.g. https://github.com/streetcomplete/StreetComplete/issues/844#issuecomment-364185029). Now, having them show only on demand is somewhat different, but it may still apply as target audience is not even expected to understand what tags _are_, much less make decision based on what is in them...

If you're feeling adventurous, the fork at https://github.com/Helium314/StreetComplete/releases/ does include settings to show most (if not all) POIs. You can install it alongside the upstream StreetComplete and see how that feature works for you (e.g. you can have profile which shows all POIs, and one which does not, and then quick-switch between them on demand).

**Comment by westnordost** [comment]
This is out of scope of this app. There are some alternatives, @Helium314 mentioned one. [OSMFocus Reborn](https://f-droid.org/packages/net.pfiers.osmfocus/) is another.

Additionally, there is room for a "Shop layer" in the future, that would show shops (etc.) on the map and let you, ... I don't know, ... state that they are still there plus add new ones or something. But this would be a different ticket, the specific feature you request is, as I wrote, out of scope.

**Comment by IzzySoft** [comment]
Well, was worth asking at least – thanks for your answer, though it wasn't the one I hoped for. And shops were just one example. Finding something else (like a bench in a park) and wondering whether to leave a note is the same scope (I stopped doing that as I cannot tell if it's already mapped or not, and I do not want to create unnecessary noise). I thought the "overlay" on a specific point and on-request might be a better choice than an entire overlay, so I suggested it.

Of course fully accepted if you consider that out-of-scope – you don't want to overload (or oversize) the app. Thanks for the alternatives mentioned, I'll take a look – especially the description of Helium given by @mnalis sounds promising (unfortunately it's not on F-Droid – the links go to your app here). I certainly won't run multiple apps in parallel; I love being able to contribute on-the-fly, and my wife already gets nervous by that, so no way to even expand to multi-task :see_no_evil:
